File: templates/fastapi-basic/app/health.py
from fastapi import APIRouter, status
from typing import Dict

router = APIRouter(
    prefix="/health",
    tags=["system"],
    responses={404: {"description": "Not found"}},
)

@router.get("",
    response_model=Dict[str, str],
    summary="Health check endpoint",
    description="Checks if the application is running properly",
    status_code=status.HTTP_200_OK,
    responses={
        200: {
            "description": "Application is healthy",
            "content": {
                "application/json": {
                    "example": {"status": "healthy"}
                }
            }
        },
        503: {
            "description": "Application is unhealthy",
            "content": {
                "application/json": {
                    "example": {"status": "unhealthy", "details": "Service unavailable"}
                }
            }
        }
    }
)
async def health_check() -> Dict[str, str]:
    """
    Performs a health check of the application.
    
    Returns:
        Dict[str, str]: A dictionary containing the health status
    """
    return {"status": "healthy"} 

File: templates/rust-axum-basic/src/health.rs
use axum::{routing::get, Json, Router};
use serde::Serialize;
use utoipa::ToSchema;

#[derive(Serialize, ToSchema)]
pub struct HealthResponse {
    status: String,
}

/// Health check endpoint handler
#[utoipa::path(
    get,
    path = "/health",
    tag = "system",
    responses(
        (status = 200, description = "System is healthy", body = HealthResponse),
        (status = 503, description = "System is unhealthy", body = HealthResponse)
    )
)]
pub async fn health_check() -> Json<HealthResponse> {
    Json(HealthResponse {
        status: "healthy".to_string(),
    })
}

pub fn router() -> Router {
    Router::new().route("/", get(health_check))
} 